src/layers/linear.rs

```diff
@@ -3,30 +3,57 @@ use crate::prelude::*;
 use crate::utils::WeightInit;
 
 pub struct Linear {
+    name: String,
     weights: Tensor<Ix2>,
     biases: Tensor<Ix1>,
 }
 
 impl Linear {
-    pub fn new(nin: usize, nout: usize) -> Self {
+    pub fn new(name: impl ToString, nin: usize, nout: usize) -> Self {
+        let name = name.to_string();
         let weights = Tensor::from_shape_simple_fn((nin, nout), || {
             WeightInit::GlorotUniform.sample([nin, nout])
         });
         let biases = Tensor::from_shape_simple_fn(nout, Value::zero);
 
-        Linear { weights, biases }
+        Linear {
+            name,
+            weights,
+            biases,
+        }
     }
 }
 
 impl Layer<Ix1, Ix1> for Linear {
-    fn parameters(&self) -> Tensor<Ix1> {
-        let mut params = self.weights.clone().into_raw_vec();
-        params.append(&mut self.biases.clone().into_raw_vec());
+    fn forward(&self, input: &Tensor<Ix1>) -> Tensor<Ix1> {
+        input.dot(&self.weights) + &self.biases
+    }
 
-        Tensor::from_vec(params)
+    fn weights(&self) -> Tensor<Ix1> {
+        Tensor::from_vec(self.weights.clone().into_raw_vec())
     }
 
-    fn forward(&self, input: &Tensor<Ix1>) -> Tensor<Ix1> {
-        input.dot(&self.weights) + &self.biases
+    fn biases(&self) -> Tensor<Ix1> {
+        Tensor::from_vec(self.biases.clone().into_raw_vec())
```

Comment:
bias is already a `Tensor<Ix1>`
```suggestion
        self.biases.clone()
```